src/pam/main.cc

```diff
@@ -155,10 +171,11 @@ int identify(pam_handle_t *pamh, int flags, int argc, const char **argv,
   posix_spawn_file_actions_init(&file_actions);
   posix_spawn_file_actions_addclose(&file_actions, STDOUT_FILENO);
   posix_spawn_file_actions_addclose(&file_actions, STDERR_FILENO);
-  const char *const args[] = {"python", "/lib/security/howdy/compare.py",
+  const char *const args[] = {"/usr/bin/python3", "/lib/security/howdy/compare.py",
                               user_ptr, nullptr};
   pid_t child_pid;
-  if (posix_spawnp(&child_pid, "python", &file_actions, nullptr,
+
+  if (posix_spawnp(&child_pid, "/usr/bin/python3", &file_actions, nullptr,
```

Comment:
We don't need to make python3 absolute since `posix_spawnp` already search the file in `$PATH` for us.

Reply:
That's what i thought, but i learned the hard way that `python3` changes in `$PATH` when in a pyenv, potentially breaking howdy. Hardcoding also fixes a vulnerability where an attacker could simply overwrite the python3 path to any executable that returns code 0 and bypass Howdy completely

Reply:
I wasn't aware of it. I reverted my commit.